zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -1355,17 +1362,37 @@ object JsonCodec {
       var i = 0
       while (i < len) {
         if (buffer(i) == null) {
-
-          if ((fields(i).optional || fields(i).transient) && fields(i).defaultValue.isDefined)
+          if ((fields(i).optional || fields(i).transient) && fields(i).defaultValue.isDefined) {
             buffer(i) = fields(i).defaultValue.get
-          else
+          } else if (config.ignoreEmptyCollections && isCollectionSchema(schemas(i))) {
+            buffer(i) = createEmptyCollection(schemas(i))
+          } else {
             buffer(i) = schemaDecoder(schemas(i)).unsafeDecodeMissing(spans(i) :: trace)
-
+          }
         }
         i += 1
       }
       buffer
     }
+
+// Helper method to check if a schema represents a collection type
+    private def isCollectionSchema(schema: Schema[_]): Boolean =
+      schema match {
+        case Schema.Sequence(_, _, _, _, _) => true
+        case Schema.Set(_, _)               => true
+        case Schema.Map(_, _, _)            => true
+        case _                              => false
+      }
+
+// Helper method to create an empty collection based on the schema
+    private def createEmptyCollection(schema: Schema[_]): Any =
```

Comment:
I think this logic belongs to the `trait Collection` in `Schema.scala`, so each collection schema has to implement a method that creates an empty collection.